nanoFramework.System.Net/Sockets/Socket.cs

```diff
@@ -336,39 +346,69 @@ public void Bind(EndPoint localEP)
         /// </remarks>
         public void Connect(EndPoint remoteEP)
         {
-            if (m_Handle == -1)
+            try
             {
-                throw new ObjectDisposedException();
-            }
+                if (m_Handle == -1) {
+                    throw new ObjectDisposedException();
+                }
 
-            EndPoint endPointSnapshot = remoteEP;
-            Snapshot(ref endPointSnapshot);
+                EndPoint endPointSnapshot = remoteEP;
+                Snapshot(ref endPointSnapshot);
 
-            if(m_fBlocking)
-            {
-                // blocking connect
-                _nonBlockingConnectInProgress = false;
-            }
-            else
-            {
-                // non blocking connect
-                _nonBlockingConnectInProgress = true;
-                _nonBlockingConnectRightEndPoint = endPointSnapshot;
-            }
-            
-            NativeSocket.connect(this, endPointSnapshot, !m_fBlocking);
+                if (m_fBlocking) {
+                    // blocking connect
+                    _nonBlockingConnectInProgress = false;
+                } else {
+                    // non blocking connect
+                    _nonBlockingConnectInProgress = true;
+                    _nonBlockingConnectRightEndPoint = endPointSnapshot;
+                }
 
-            if (m_fBlocking)
-            {
-                // if we are on blocking connect
-                // poll until connection is established or exception thrown
-                Poll(-1, SelectMode.SelectWrite);
-            }
+                //NativeSocket.connect(this, endPointSnapshot, !m_fBlocking);
 
-            if (_rightEndPoint == null)
+                //if (m_fBlocking) {
+                //    // if we are on blocking connect
+                //    // poll until connection is established or exception thrown
+                //    Poll(-1, SelectMode.SelectWrite);
+                //}
+
+                // skigrinder - 2020-12-10 added exception handling and timeout for Poll()
+                try
+                {
+                    NativeSocket.connect(this, endPointSnapshot, !m_fBlocking);
+                }
+                catch (Exception ec)
+                {
+                    Debug.WriteLine($"System.Net - Connect() - NativeSocket.connect() - Exception caught {ec.GetType().Name} ");
+                }
+
+                try
+                {
+                    if (m_fBlocking)
+                    {
+                        // if we are on blocking connect
+                        //Poll(-1, SelectMode.SelectWrite);                                 // skigrinder - old code would poll until connection was established or exception thrown - this hung nanoFramework applications
+                        if (!Poll(1500000, SelectMode.SelectWrite))                         // 'microseconds' values (1st param) - 1500000 is 1.5 seconds;
```

Comment:
If the socket is set to blocking, the call is supposed to be blocked until the connection happens.
The way it's implemented it shouldn't prevent the other threads from running. If that's the case than it's an issue with the interpreter not here.

Reply:
Can we leave this timeout in until the interpreter is fixed??
Hanging the application here causes the .Net & .Net.Http libraries to be unusable.

Reply:
How's that? The -1 here is exactly to make the execution stop until a connection is established (or the default timeout occurring)... Isn't that happening now?

Reply:
No.  My applications were hanging here.  But I haven't tested the -1 option since the recent  interpreter/extension changes were made.